src/main/java/org/apache/flink/benchmark/ContinuousFileReaderOperatorBenchmark.java

```diff
@@ -0,0 +1,118 @@
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+
+@OperationsPerInvocation(value = ContinuousFileReaderOperatorBenchmark.RECORDS_PER_INVOCATION)
+public class ContinuousFileReaderOperatorBenchmark extends BenchmarkBase {
+    private static final int SPLITS_PER_INVOCATION = 100;
+    private static final int LINES_PER_SPLIT = 20_000_000;
+    public static final int RECORDS_PER_INVOCATION = SPLITS_PER_INVOCATION * LINES_PER_SPLIT;
+
+    private static final TimestampedFileInputSplit SPLIT = new TimestampedFileInputSplit(0, 0, new Path("."), 0, 0, new String[]{});
+    private static final String LINE = Strings.repeat('0', 10);
+
+    public static void main(String[] args)
+            throws RunnerException {
+        Options options = new OptionsBuilder()
+                .verbosity(VerboseMode.NORMAL)
+                .include(".*" + ContinuousFileReaderOperatorBenchmark.class.getSimpleName() + ".*")
+                .build();
+
+        new Runner(options).run();
+    }
+
+    @Benchmark
+    public void readFiles(FlinkEnvironmentContext context) throws Exception {
+        StreamExecutionEnvironment env = context.env;
+        env
+                .enableCheckpointing(100)
+                .setParallelism(1)
+                .addSource(buildSourceFun())
+                .transform("fileReader", TypeInformation.of(String.class),
+                        new ContinuousFileReaderOperator<>(new MockInputFormat()))
+                .addSink(new DiscardingSink<>());
+
+        env.execute();
+    }
+
+    private SourceFunction<TimestampedFileInputSplit> buildSourceFun() {
+        return new SourceFunction<TimestampedFileInputSplit>() {
+            private volatile boolean isRunning = true;
+            private int count = 0;
+
+            @Override
+            public void run(SourceContext<TimestampedFileInputSplit> ctx) {
+                while (isRunning && count < SPLITS_PER_INVOCATION) {
+                    count++;
+                    synchronized (ctx.getCheckpointLock()) {
+                        ctx.collect(SPLIT);
+                    }
+                }
+            }
+
+            @Override
+            public void cancel() {
+                isRunning = false;
+            }
+        };
+    }
+
+    private static class MockInputFormat extends FileInputFormat<String> {
+        private transient int count = 0;
+
+        @Override
+        public boolean reachedEnd() {
+            return count >= ContinuousFileReaderOperatorBenchmark.LINES_PER_SPLIT;
+        }
+
+        @Override
+        public String nextRecord(String s) {
+            count++;
+            return LINE;
+        }
+
+        @Override
+        public void open(FileInputSplit fileSplit) {
+            // prevent super from accessing file
```

Comment:
missing 
```
count = 0;
```
As it is now, the throughput is inflated 100 times 👿 Probably the the `LINES_PER_SPLIT` should be decreased as a side effect to `200_000`?